src/main/kotlin/com/openlattice/assembler/tasks/MaterializePermissionSyncTask.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.assembler.tasks
+
+import com.hazelcast.query.Predicate
+import com.hazelcast.query.Predicates
+import com.openlattice.assembler.EntitySetAssemblyKey
+import com.openlattice.assembler.MaterializedEntitySetsDependencies
+import com.openlattice.authorization.Permission
+import com.openlattice.authorization.PrincipalType
+import com.openlattice.organization.OrganizationEntitySetFlag
+import com.openlattice.postgres.PostgresColumn
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable.MATERIALIZED_ENTITY_SETS
+import com.openlattice.postgres.PostgresTable.PERMISSIONS
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.mapstores.MaterializedEntitySetMapStore
+import com.openlattice.postgres.streams.PostgresIterable
+import com.openlattice.postgres.streams.StatementHolder
+import com.openlattice.tasks.HazelcastFixedRateTask
+import com.openlattice.tasks.Task
+import java.sql.ResultSet
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+import java.util.function.Supplier
+
+
+private val selectUnauthorizedEntitySetAssemblies =
+        "SELECT ${ENTITY_SET_ID.name}, ${ORGANIZATION_ID.name} FROM ${MATERIALIZED_ENTITY_SETS.name} " +
```

Comment:
Does this not need to be computed for each organization? e.g. if an entity set is materialized by two organizations, and one of those organizations has lost its MATERIALIZE permissions, it should still be marked as needing removal from one organization's database? (or does this delete not affect individual organization databases?)

Reply:
For entity set permission changes in materialization, we don't need to mark them, because: 

- if permissions get's added to an organizations, there is no action required (it just means, that owners can materialize the entity set if they choose so) as opposed to, when materialization permission get's added to a property type, that is part of a materialized entity set, we need to re-create the materialized view with the new column
- if permission get's removed, we don't need to mark it for entity sets, because they can be picked up with this query and get deleted from that organizations database (and from mapstore)

Reply:
Right, but if an entity set is materialized in two organizations' databases, and only one of them loses its MATERIALIZE permissions, then it won't get picked up by this query, right?

Reply:
Ohh, you're right! There would need to be a join on organization too, but that's not possible from these 2 tables. I will mark entity sets as well then.